rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -258,6 +258,29 @@ inline void microphone_device::variable_byteswap(const void* src, void* dst)
 	}
 }
 
+inline u32 microphone_device::convert_16_bit_pcm_to_float(const std::vector<u8>& buffer, usz num_bytes)
+{
+	constexpr usz float_buf_size = 2 * inbuf_size;
+	static_assert((float_buf_size % sizeof(u32)) == 0);
+
+	float_buf.resize(float_buf_size, 0);
+	ensure(num_bytes * 2 <= float_buf.size());
+
+	const be_t<s16>* src = reinterpret_cast<const be_t<s16>*>(static_cast<const void*>(buffer.data()));
+	be_t<f32>* dst = reinterpret_cast<be_t<f32>*>(float_buf.data());
+
+	for (usz i = 0; i < num_bytes / 2; i++)
```

Comment:
I think we should use sizeof to make this less magical
```suggestion
	for (usz i = 0; i < num_bytes / sizeof(s16); i++)
```